libraries/go/webhook.go

```diff
@@ -127,13 +194,30 @@ func (wh *Webhook) Sign(msgId string, timestamp time.Time, payload []byte) (stri
 
 func (wh *Webhook) sign(msgId string, timestamp time.Time, payload []byte) (version string, signature []byte, err error) {
 	toSign := fmt.Sprintf("%s.%d.%s", msgId, timestamp.Unix(), payload)
+	switch wh.signingMethod {
+	// TODO: conver this to constant
+	case HMAC:
+		h := hmac.New(sha256.New, wh.key)
+		h.Write([]byte(toSign))
+		sig := h.Sum(nil)
+		base64Sig := make([]byte, base64.StdEncoding.EncodedLen(len(sig)))
+		base64.StdEncoding.Encode(base64Sig, sig)
+		return "v1", base64Sig, nil
+
+	case ED25519:
+		if ed25519.PrivateKey(wh.privateKey).Seed() == nil {
```

Comment:
All random strings are valid Ed25519 private keys. I don't think this is correct.